src/util/ALE.sol

```diff
@@ -0,0 +1,235 @@
+//SPDX-License-Identifier: None
+pragma solidity ^0.8.0;
+
+import "src/interfaces/IMarket.sol";
+import "src/interfaces/IERC20.sol";
+import "src/interfaces/IDola.sol";
+import "src/interfaces/ITransformHelper.sol";
+import {Ownable} from "lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+import {ReentrancyGuard} from "lib/openzeppelin-contracts/contracts/security/ReentrancyGuard.sol";
+
+// Accelerated leverage engine
+contract ALE is Ownable, ReentrancyGuard {
+    error CollateralNotSet();
+    error MarketNotSetForCollateral(address collateral);
+    error SwapFailed();
+    error DOLAInvalidBorrow();
+    error DOLAInvalidRepay();
+    error InvalidProxyAddress();
+
+    // 0x ExchangeProxy address.
+    // See https://docs.0x.org/developer-resources/contract-addresses
+    address payable public exchangeProxy;
+
+    IDola public immutable dola;
+
+    struct Market {
+        address market;
+        address helper;
```

Comment:
```suggestion
        IMarket market;
        ITransformHelper helper;
```

Reply:
Fixed in [21265a2](https://github.com/InverseFinance/FiRM/pull/52/commits/21265a282ef6b19fa53c46e5b48fe83f48880686)